main.cpp

```diff
@@ -82,23 +88,22 @@ int main()
                 {
                     guesses = 0;
                     guessedchar.clear();
-                    srand(time(NULL));
                     int random = rand() % list.size(); // randomise the word
                     string word = list[random];
                     printoptions();
                     break;
                 }
                 else if (input == 'n' || input == 'N')
                 {
-                    return 0;
+                    isGameRunning = false;
```

Comment:
indeed, you're doing it already. the main loop can only check for the boolean, this line should take care of stopping the loop if the player wants to stop playing